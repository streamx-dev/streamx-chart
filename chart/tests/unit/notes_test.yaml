# Copyright 2024 Dynamic Solutions Sp. z o.o. sp.k.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json

suite: test chart notes
templates:
  - templates/NOTES.txt
tests:
  # @Test
  - it: when rest_ingestion service ingress host is set then it is printed in notes with http
    set:
      rest_ingestion:
        ingress:
          host: rest.ingestion.host
    asserts:
      - matchSnapshot: {}
      - matchRegexRaw:
          pattern: http://rest.ingestion.host
  # @Test
  - it: when rest_ingestion service ingress tls is set then it is printed in notes with https
    set:
      rest_ingestion:
        ingress:
          host: rest.ingestion.host
          tls:
            secretName: tls-secret
    asserts:
      - matchSnapshot: {}
      - matchRegexRaw:
          pattern: https://rest.ingestion.host
  # @Test
  - it: when delivery service ingress enabled then host printed in notes
    set:
      delivery:
        service1:
          outputs:
            - name: output1
              service:
                port: 80
                targetPort: http
              ingress:
                enabled: true
                host: service1.host
                path: /testpath
    asserts:
      - matchSnapshot: {}
