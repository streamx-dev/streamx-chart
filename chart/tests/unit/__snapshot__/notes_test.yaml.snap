when chart is installed and messaging.pulsar is configured then NOTES contains initialisation message:
  1: |
    raw: |-
      # Copyright 2024 Dynamic Solutions Sp. z o.o. sp.k.
      #
      # Licensed under the Apache License, Version 2.0 (the "License");
      # you may not use this file except in compliance with the License.
      # You may obtain a copy of the License at
      #
      #     http://www.apache.org/licenses/LICENSE-2.0
      #
      # Unless required by applicable law or agreed to in writing, software
      # distributed under the License is distributed on an "AS IS" BASIS,
      # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
      # See the License for the specific language governing permissions and
      # limitations under the License.

      Happy StreamX Helming!
      ---------------------
      Apache Pulsar initialization started...
      Please check the status with:

      kubectl -n my-namespace wait --for=condition=complete job --selector app.kubernetes.io/component=pulsar-init-tenant --timeout=300s

      Deploy StreamX Mesh when the initialization completes successfully.
when delivery service ingress configured then host printed in notes:
  1: |
    raw: |-
      # Copyright 2024 Dynamic Solutions Sp. z o.o. sp.k.
      #
      # Licensed under the Apache License, Version 2.0 (the "License");
      # you may not use this file except in compliance with the License.
      # You may obtain a copy of the License at
      #
      #     http://www.apache.org/licenses/LICENSE-2.0
      #
      # Unless required by applicable law or agreed to in writing, software
      # distributed under the License is distributed on an "AS IS" BASIS,
      # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
      # See the License for the specific language governing permissions and
      # limitations under the License.

      Happy StreamX Helming!
      ---------------------
      Delivery services:
      - http://service1.host
when delivery service ingress configured with tls then host printed in notes with https:
  1: |
    raw: |-
      # Copyright 2024 Dynamic Solutions Sp. z o.o. sp.k.
      #
      # Licensed under the Apache License, Version 2.0 (the "License");
      # you may not use this file except in compliance with the License.
      # You may obtain a copy of the License at
      #
      #     http://www.apache.org/licenses/LICENSE-2.0
      #
      # Unless required by applicable law or agreed to in writing, software
      # distributed under the License is distributed on an "AS IS" BASIS,
      # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
      # See the License for the specific language governing permissions and
      # limitations under the License.

      Happy StreamX Helming!
      ---------------------
      Delivery services:
      - https://service1.host
when rest_ingestion service ingress host is set then it is printed in notes with http:
  1: |
    raw: |-
      # Copyright 2024 Dynamic Solutions Sp. z o.o. sp.k.
      #
      # Licensed under the Apache License, Version 2.0 (the "License");
      # you may not use this file except in compliance with the License.
      # You may obtain a copy of the License at
      #
      #     http://www.apache.org/licenses/LICENSE-2.0
      #
      # Unless required by applicable law or agreed to in writing, software
      # distributed under the License is distributed on an "AS IS" BASIS,
      # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
      # See the License for the specific language governing permissions and
      # limitations under the License.

      Happy StreamX Helming!
      ---------------------
      Ingestion services:
      - http://rest.ingestion.host/q/swagger-ui
when rest_ingestion service ingress tls is set then it is printed in notes with https:
  1: |
    raw: |-
      # Copyright 2024 Dynamic Solutions Sp. z o.o. sp.k.
      #
      # Licensed under the Apache License, Version 2.0 (the "License");
      # you may not use this file except in compliance with the License.
      # You may obtain a copy of the License at
      #
      #     http://www.apache.org/licenses/LICENSE-2.0
      #
      # Unless required by applicable law or agreed to in writing, software
      # distributed under the License is distributed on an "AS IS" BASIS,
      # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
      # See the License for the specific language governing permissions and
      # limitations under the License.

      Happy StreamX Helming!
      ---------------------
      Ingestion services:
      - https://rest.ingestion.host/q/swagger-ui
